nautobot/extras/models/groups.py

```diff
@@ -431,3 +446,199 @@ def clean(self):
+        # return DynamicGroup.objects.filter(pk__in=descendants)
+
+    def get_ancestors(self, group=None, ancestors=None):
+        """Return the parents of all parent groups."""
+        if group is None:
+            group = self
+        if ancestors is None:
+            ancestors = []
+
+        for parent_group in group.parents.all():
+            print(f"Processing group {parent_group}...")
+            # ancestors.append(parent_group.pk)
+            ancestors.append(parent_group)
+            if parent_group.parents.exists():
+                self.get_ancestors(parent_group, ancestors)
+
+        # TODO(jathan): Return a QuerySet with default ordering? Or a list that
+        # is explicitly ordered by distance from object.
+        return ancestors
+        # return DynamicGroup.objects.filter(pk__in=ancestors)
+
+    def get_siblings(self, parent_group=None):
+        """Return groups that share the same parents."""
+        if parent_group is None:
+            parent_group = self
+
+        return DynamicGroup.objects.filter(parents=parent_group)
+
+    def is_leaf(self):
+        """Return whether this is a leaf node (has no children)."""
+        return not self.children.exists()
+
+
+def limit_dynamic_group_by_parent_content_type():
+    """
+    Callback for use on `DynamicGroupMembership.group.limit_choices_to`.
+
+    This asserts that only Dynamic Groups matching the `content_type` of the `parent_group` can be selected.
+    """
+    return models.Q(parents__content_type=models.F("content_type"))
+
+
+class DynamicGroupMembership(BaseModel):
+    """Intermediate model for associating filters to groups."""
+
+    group = models.ForeignKey(
+        "extras.DynamicGroup",
+        on_delete=models.CASCADE,
+        related_name="+",
+        limit_choices_to=limit_dynamic_group_by_parent_content_type,
+    )
+    parent_group = models.ForeignKey(
+        "extras.DynamicGroup", on_delete=models.CASCADE, related_name="dynamic_group_memberships"
+    )
+    operator = models.CharField(choices=DynamicGroupOperatorChoices.CHOICES, max_length=12)
+    weight = models.PositiveSmallIntegerField()
+
+    objects = DynamicGroupMembershipQuerySet.as_manager()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["group", "parent_group", "operator", "weight"], name="group_to_filter_uniq"
+            ),
+        ]
+        ordering = ["parent_group", "weight", "group"]
+
+    def __str__(self):
+        return f"{self.group}: {self.operator} ({self.weight})"
+
+    def natural_key(self):
+        return self.group.natural_key() + self.parent_group.natural_key() + (self.operator, self.weight)
+
+    natural_key.dependencies = ["extras.dynamicgroup"]
+
+    @property
+    def name(self):
+        return self.group.name
+
+    @property
+    def members(self):
+        return self.group.members
+
+    @property
+    def count(self):
+        return self.group.count
```

Comment:
I added them for the purpose of tables/UI so that we can extract the membership information (`operator`, `weight`) and still directly access the underlying attributes from the associated `group` without any other hacks.  I mentioned it in the commit comments.